slack/webhook/async_client.py

```diff
@@ -0,0 +1,173 @@
+        ssl: Optional[SSLContext] = None,
+        proxy: Optional[str] = None,
+        trust_env_in_session: bool = False,
+        session: Optional[aiohttp.ClientSession] = None,
+        auth: Optional[BasicAuth] = None,
+        default_headers: Optional[Dict[str, str]] = None,
+    ):
+        """API client for Incoming Webhooks and response_url
+        :param url: a complete URL to send data (e.g., https://hooks.slack.com/XXX)
+        :param timeout: request timeout (in seconds)
+        :param ssl: ssl.SSLContext to use for requests
+        :param proxy: proxy URL (e.g., localhost:9000, http://localhost:9000)
+        :param default_headers: request headers to add to all requests
+        """
+        self.url = url
+        self.timeout = timeout
+        self.loop = loop
+        self.ssl = ssl
+        self.proxy = proxy
+        self.trust_env_in_session = trust_env_in_session
+        self.session = session
+        self.auth = auth
+        self.default_headers = default_headers if default_headers else {}
+
+    async def send(
+        self,
+        *,
+        text: Optional[str] = None,
+        attachments: Optional[List[Union[Dict[str, any], Attachment]]] = None,
+        blocks: Optional[List[Union[Dict[str, any], Block]]] = None,
+        response_type: Optional[str] = None,
+        headers: Optional[Dict[str, str]] = None,
+    ) -> WebhookResponse:
+        """Performs a Slack API request and returns the result.
+        :param text: the text message (even when having blocks, setting this as well is recommended as it works as fallback)
+        :param attachments: a collection of attachments
+        :param blocks: a collection of Block Kit UI components
+        :param response_type: the type of message (either 'in_channel' or 'ephemeral')
+        :param headers: request headers to append only for this request
+        :return: API response
+        """
+        return await self.send_dict(
+            body={
+                "text": text,
+                "attachments": attachments,
+                "blocks": blocks,
+                "response_type": response_type,
+            },
+            headers=headers,
+        )
+
+    async def send_dict(
+        self, body: Dict[str, any], headers: Optional[Dict[str, str]] = None
+    ) -> WebhookResponse:
+        """Performs a Slack API request and returns the result.
+        :param body: json data structure (it's still a dict at this point),
+            if you give this argument, body_params and files will be skipped
+        :param headers: request headers to append only for this request
+        :return: API response
+        """
+        body = {k: v for k, v in body.items() if v is not None}
+        body = convert_bool_to_0_or_1(body)
+        _parse_web_class_objects(body)
+        return await self._perform_http_request(
+            body=body, headers=self._build_request_headers(headers)
+        )
+
+    async def _perform_http_request(
```

Comment:
mostly cut and paste from `WebhookClient`. The main changes are:
* Make methods async
* Use aiohttp in this method